File: hackerearth/multiple-occurrences.cpp
/**
 * https://www.hackerearth.com/practice/basic-programming/implementation/basics-of-implementation/practice-problems/algorithm/multiple-occurence-97c00160/
 **/
#include<iostream>
#include<unordered_map>
using namespace std;

int main() {
	ios_base::sync_with_stdio(false);
	int t, n, x;
	cin>>t;
	while(t--) 
	{
		cin>>n;
		unordered_map<int, pair<int, int>> freq; 
		for(int i=0; i<n; i++) 
		{
			cin >> x;
			if (freq.find(x) == freq.end()) {
				freq[x] = {i, i};
			} else {
				freq[x].second = i;
			}
		}

		int ans = 0;
		for(auto item: freq) {
			ans += item.second.second - item.second.first;
		}
		cout<<ans<<"\n";
	}
}

File: hackerearth/java/MultipleOccurrences.java
/**
 * https://www.hackerearth.com/practice/basic-programming/implementation/basics-of-implementation/practice-problems/algorithm/multiple-occurence-97c00160/
 **/
import java.util.*;

class Tuple {
    int low;
    int high;

    Tuple(int l, int h) {
        low=l;
        high=h;
    }
}

class MultipleOccurrences {
    public static void main(String[] args) throws Exception {
        int x, n, t;
        Scanner s = new Scanner(System.in);
        Map<Integer, Tuple> freq;

        t = s.nextInt();
        while(t > 0) {
            s.nextLine();
            freq = new HashMap<>();

            n = s.nextInt(); s.nextLine();
            for (int i = 0; i < n; i++) {
                x = s.nextInt();
                if (freq.containsKey(x)) {
                    Tuple val = freq.get(x);
                    val.high = i;
                } else {
                    freq.put(x, new Tuple(i, i));
                }
            }
            int ans = freq.values()
                    .stream()
                    .map(value -> value.high - value.low)
                    .reduce(0, Integer::sum);
            System.out.println(ans);
            
            t--;
        }
        s.close();
    }
}